### PR TITLE
Monorepo the `forth3` project

### PR DESCRIPTION
**Pull request by jamesmunns**
Keeps `forth3` and `f3repl`, `forthboy`'s parts now live in https://github.com/tosc-rs/teletype/.

Same deal as #54.

This should NOT be a squash merge.

### CODE REVIEW

File: source/forth3/.github/workflows/ci.yml
Comment (hawkw) [inline source/forth3/.github/workflows/ci.yml:1]: we'll presumably need to delete this and roll it into the main CI config

File: source/forth3/.github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,165 @@
+name: CI
```
Comment (jamesmunns) [inline source/forth3/.github/workflows/ci.yml:1]: This should eventually be integrated into mnemos' CI. It's pretty important IMO to preserve regular + miri testing.

**Review by hawkw — APPROVED** [review]
we should probably remove the separate CI config in forth3, because, AFAIK, a `.github` dir that isn't in the repo root doesn't do anything. other than that, LGTM!

File: source/forth3/.github/workflows/ci.yml
Comment (jamesmunns) [reply source/forth3/.github/workflows/ci.yml:1]: Opened as #58!

**Pull request merged**